### ISSUE TITLE
Response from Amazon on MFN createShipment is not conform and results in an exception

### ISSUE THREAD

**Issue by Inbasecom** [jlevers/selling-partner-api#200]
In the MerchantFullfilmentApi endpoint createShipment, a response from the SP API is only processed if it matches the JSON model. We observe that Amazon often provides too long values in the "Name" field of the "Address".
Processing of the response is then no longer possible, because the model "Address" is not filled and only throws an exception.

If you change the value to 60 in the model https://github.com/jlevers/selling-partner-api/blob/main/lib/Model/MerchantFulfillment/Address.php in the function "setName" in line 339 (if query), the response is processed correctly. Amazon itself defines a maximum length of the field with 30 characters.
However, a majority of orders come with a response much longer than 30 characters in the field. Truncating the field does not make sense, because the address is immensely important for the shipping route.

We have already opened a case on this with Amazon Developer Support, however we have been waiting for a long time now for a more advanced response.

Any ideas to solve it for now?

**Comment by jlevers** [comment]
This should be fixed in `v4.3.1` -- give it a try and let me know if you still run into issues :)

**Comment by Inbasecom** [comment]
Hello,

yes that is how it works.
But the problem should actually be fixed on Amazon side. Do you see it the same way?
Amazon defines in its own model, 30 characters, but passes more than 30 characters. The same model is also used to pass the sender to Amazon. There Amazon rejects a value with 30 characters but generally.

Nevertheless, thanks for the fix.

**Comment by jlevers** [comment]
I agree that it should be fixed on the Amazon end, but I have not had any luck getting Amazon to update bugs in their OpenAPI models, so I just maintain a fork of their models that I use for generating this library. If you'd like to try to contact them to get htem up update their models, I'd open an issue in their [models repository](https://github.com/amzn/selling-partner-api-models/).
